ClinGen/clincoded
# Bug Fix #2066: Fixes Semidominant publishing issue

#2066 
Users are now able to Publish Classifications under the Semidominant MOI. Added Semidominant to `allowPublishGlobal`, users will no longer receive an alert that this is not permitted.

Steps to test (VCI):
- Curate a new variant, and click on "Interpretation+" button
- Click "Inheritance+" button and select "Semidominant" from mode of inheritance dropdown, click Ok
- Choose a criteria to evaluate for (i.e. PP4), and save the evaluation.
- Navigate to view summary, and save until getting to "Approve Interpretation"
- Originally, this bug would not allow users to Publish their Interpretations, as Semidominant MOI had not been included in the VCI schema handling Approvals/Publish, as described by Matt here: #2066 
- The error/warning message should no longer appear, and users can publish using Semidominant MOI (must be part of an approved VCEP to allow publishing)

Steps to test (GCI):
Original PR here: https://github.com/ClinGen/clincoded/pull/1937
- In "create-gene-disease" -> under "Mode of Inheritance" pull down -> "Semidominant inheritance (HP:0032113)"

**Individual Curation**
- "Curate Individual Information" -> Under "Individual - Associate Variant(s) -> "The proband is:" question should be visible with 4 choices highlighted in spec (required question), answer should be visible in Evidence Summary under Variant.
- If answer to above question is "Biallelic homozygous" or "Biallelic compound heterozygous", 3 questions pop under "Individual - Associate Variants": 
   - If there are 2 variants described, are they both located in _trans_ with respect to one another?
   - If the individual has one variant, is it _de novo_ OR If the individual has 2 variants, is at least one _de novo_?
   - If the answer to the above _de novo_ question is yes, is the variant maternity and paternity confirmed?
- Under "Individual - Score Proband" -> "Select Status" -> Score -> "Confirm Case Information Type" -> 5 scoring options are available in drop-down

**Family Curation**
 - Under Family -> Tested Individuals -> First question should be "Published LOD score?" with respective dropdown selections.

 - If answer to above question is "Yes", scoring input questions are grayed out, onward to the following question below: "Which mode of inheritance does this family display?"
    - If Semidominant, "Does the family meet requirements for estimating LOD score for EITHER autosomal dominant/X-linked or autosomal recessive?"
    - Depending on answer above, inputs should also gray out in correspondence to MOI "rules" (similar to original family MOI question above).

 - Inputs used for calculation below the above question should gray out depending on answer. eLOD input should also be locked. LOD Calculations based on input should function as normal.

 - Under "Family - Variants segregating with proband" -> "Proband label" -> Ability to add 3 variants instead of two.

**Classification Matrix**
 - For Semidominant inheritance scoring and evidence should be appropriately represented, maxed out at 12 points for "Genetic Evidence Total".

**Evidence Summary Overview**
 - "Proband is:" info present under Variant
 - if non-proband family is curated and scored, column should be sortable based on MOI (Genetic Evidence: Case Level (family segregation information without proband data or scored proband data)
 - No text overflow on right side of tables
